refactor: Rename huddle to direct message group in web codebase.

roanster007 opened the pull request:
This commit renames "huddle" to "direct_message_group" in the web codebase. It also renames the file - "huddle_data.ts" to "direct_message_group_data.ts"

Fixes part of #28640

<!-- Describe your pull request here.-->

Fixes: <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
Can you rename the file in a separate commit from renaming the variables? It's really hard to review a file rename squashed with internal changes to that file.

timabbott commented on web/src/filter.ts:
```diff
@@ -727,7 +727,7 @@ export class Filter {
     }
 
-    is_non_huddle_pm(): boolean {
+    is_non_direct_message_group_pm(): boolean {
```
Let's call this `is_non_group_direct_message`, or maybe there was some other name discussed in Zulip for the concept for a 1:1 direct message.

`pm` is an old reference to "private messages" , and not helpful.

roanster007 replied:
I could not find other names discussed on CZO. So I renamed to `is_non_group_direct_message`.

timabbott commented on web/src/pm_conversations.ts:
```diff
@@ -19,8 +19,8 @@ export function is_partner(user_id: number): boolean {
+    // message groups in which all participants are muted.
     const recipients = people.split_to_ints(conversation.user_ids_string);
 
     if (recipients.every((id) => muted_users.is_user_muted(id))) {
```
Maybe a good follow-up project would be renaming this module to `direct_messages.ts` while we're working on this sweep? Probably worth posting a proposal in #frontend with suggested names for each of the functions that have `pm` or `private` in them before doing the work, since 90% of review will be bikeshedding the names.

roanster007 replied:
sure!

roanster007 commented:
>Can you rename the file in a separate commit from renaming the variables? It's really hard to review a file rename squashed with internal changes to that file.

Updated!

>Let's call this is_non_group_direct_message, or maybe there was some other name discussed in Zulip for the concept for a 1:1 direct message.

updated.

timabbott commented:
Merged, thanks @roanster007!
